aptos-move/block-executor/src/code_cache_global.rs

```diff
@@ -120,15 +127,51 @@ where
         self.module_cache.len()
     }
 
+    /// Returns the number of layout entries in the cache.
+    pub fn num_cached_layouts(&self) -> usize {
+        self.struct_layouts.len()
+    }
+
     /// Returns the sum of serialized sizes of modules stored in cache.
     pub fn size_in_bytes(&self) -> usize {
         self.size
     }
 
-    /// Flushes the module cache.
+    /// Flushes all caches.
     pub fn flush(&mut self) {
         self.module_cache.clear();
         self.size = 0;
+        self.struct_layouts.clear();
+    }
+
+    /// Flushes only layout caches.
+    pub fn flush_layout_cache(&self) {
+        // TODO(layouts):
+        //   Flushing is only needed because of enums. Once we refactor layouts to store a single
+        //   variant instead, this can be removed.
+        self.struct_layouts.clear();
+    }
+
+    /// Returns layout entry if it exists in global cache.
+    pub(crate) fn get_struct_layout_entry(&self, key: &StructKey) -> Option<LayoutCacheEntry> {
```

Comment:
Wonder why this isn't unified with store using `dashmap::Entry` and some lambda creating a new entry if needed? Probably marginal advantage.